### PR TITLE
Oauth

### PR DESCRIPTION
**Pull request by ShreyamsJain**
Tested manually. Facebook login is working. Please review changes

### CODE REVIEW

**Comment by ShreyamsJain** [comment]
Updated the environment variables(facebook tokens) in Codeship.

**Comment by ShreyamsJain** [comment]
Resolved the conflicts. Ready for review

**Comment by erinmorrissey** [comment]
@ShreyamsJain I'm trying to test the Facebook login — when I click "Sign In with Facebook", I get redirected to this FB error page: https://www.facebook.com/v2.6/dialog/oauth?client_id&display=popup&redirect_uri=http%3A%2F%2Flocalhost%3A3030%2Fusers%2Fauth%2Ffacebook%2Fcallback&response_type=code&scope=email&state=9c905c8bdb4b36d03f2bd7bcffdb20fc0086c1ee2af3f71c

Server log gives this:
`Started GET "/users/auth/facebook" for 10.0.2.2 at 2016-09-08 22:26:07 +0000
Cannot render console from 10.0.2.2! Allowed networks: 127.0.0.1, ::1, 127.0.0.0/127.255.255.255
I, [2016-09-08T22:26:07.100748 #4182]  INFO -- omniauth: (facebook) Request phase initiated.`

I'm still googling to see what it could be, just thought I'd bring it up in case you happened to encounter it previously?

**Comment by ShreyamsJain** [comment]
@erinmorrissey I was facing this issue yesterday, i just dont remember how i fixed it. When you are redirected to the facebook page, are you getting the error "invalid app id"?. Meanwhile i will search and let you know if how i fixed this error.

**Comment by lasley** [comment]
only the secret should go in environment vars. You'll still need the `FACEBOOK_APP_ID` var added to your omniauth middleware in the same way you did before the env vars

**Comment by ShreyamsJain** [comment]
Made the necessary changes(Added APP_ID to devise.rb) and resolved the conflicts.

**Comment by erniejamison** [comment]
Might be working but I'm getting this error from facebook:

App Not Setup: This app is still in development mode, and you don't have access to it. Switch to a registered test user or ask an app admin for permissions.

**Comment by ShreyamsJain** [comment]
Sorry for the late response @erniejamison . I checked in facebook settings and there is an option to add developers and testers. If you could tell me your username on Facebook, i can add you as a developer. I suppose, you have to be in my friend list to be added as a developer. Send me a request. My name is Shreyams Jain on fb.

**Comment by lasley** [comment]
@ShreyamsJain - The app should be in production mode for our production server. This should not require whitelist

**Comment by ShreyamsJain** [comment]
@lasley Ok. There was an option to add developers and testers, so i thought probably the app is in development mode and requires to grant access. There was another option to make the app public and i just turned it on. Maybe it should work now for Ernie.

**Comment by lasley** [comment]
@ShreyamsJain - The implementation looks good, but we will need to fix the build errors + add tests
- https://github.com/omniauth/omniauth/wiki/Integration-Testing
- https://gist.github.com/kinopyo/1338738
- https://gist.github.com/spyou/1200365

**Comment by ShreyamsJain** [comment]
@erniejamison @erinmorrissey  The files i have been working with are sessions_controller_spec.rb, rails_helper.spec, factories/omniauth.rb. Please ignore support/omniauth_macros.rb

File: app/models/pieces/untitled.Appraisals [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class King < Piece
```
Comment (lasley) [inline app/models/pieces/untitled.Appraisals]: What is this file?

File: .rubocop.yml [hunk 2]
```diff
@@ -7,3 +7,6 @@ AllCops:
 # Metrics
 Metrics/LineLength:
   Max: 100
+
+Style/BracesAroundHashParameters:
```
Comment (lasley) [inline .rubocop.yml]: Why this change?

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -26,4 +26,5 @@
   </div>
 <% end %>
 
+
```
Comment (lasley) [inline app/views/devise/registrations/new.html.erb]: Why this change?

File: coverage/.last_run.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
```
Comment (lasley) [inline coverage/.last_run.json]: Remove this file

File: coverage/.resultset.json [hunk 5]
```diff
@@ -1,7 +1,1552 @@
 {
```
Comment (lasley) [inline coverage/.resultset.json]: Remove this file

File: coverage/assets/0.10.0/application.css [hunk 6]
```diff
@@ -0,0 +1,799 @@
+/* -----------------------------------------------------------------------
```
Comment (lasley) [inline coverage/assets/0.10.0/application.css]: Remove this file

File: coverage/.last_run.json (hunk 4 above)
Comment (lasley) [reply coverage/.last_run.json]: (this whole coverage folder needs to be deleted)

**Review by lasley — CHANGES_REQUESTED** [review]

**Comment by lasley** [comment]
Hey @ShreyamsJain - we have a conflict on the controller spec due to my #38. Will need to resolve please.

**Comment by ShreyamsJain** [comment]
Hi @lasley , i updated my mac os to Sierra and after that i am unable to launch vagrant. Working on fixing that issue. Will resolve the above conflicts as soon as i get vagrant back working.

**Comment by ShreyamsJain** [comment]
@lasley This conflicts are resolved.

File: .gitignore [hunk 7]
```diff
@@ -12,6 +12,9 @@
 !/log/.keep
 /tmp
 
+# Ignore application configuration
+/config/application.yml
```
Comment (lasley) [inline .gitignore]: Why?

**Review by lasley — CHANGES_REQUESTED** [review]

**Comment by ShreyamsJain** [comment]
Hi @lasley . This has been removed from gitignore

**Review by lasley — APPROVED** [review]

**Comment by erinmorrissey** [comment]
I was going to approve and merge this, but the codecov scores are preventing it, at reasonable diff scores for this particular PR, I would say. I saw in codecov.yml where we manually set a coverage range of 80-100, but don't see if there's a GitHub setting that would cause a 90% score to cause a failing check...?

**Comment by lasley** [comment]
You can alway override the checks with admin rights. The issue is the diff, not the total. 10% diff from target will pretty much never be considered acceptable. There's a setting in the Codecov YML but I can't remember it TBH

image

**Pull request merged**